src/build123d/geometry.py

```diff
@@ -2092,18 +2093,20 @@ def optarg(kwargs, name, args, index, default):
         elif arg_face:
             # Determine if face is planar
             surface = BRep_Tool.Surface_s(arg_face.wrapped)
-            if not isinstance(surface, Geom_Plane):
+            is_surface_planar = GeomLib_IsPlanarSurface(surface, TOLERANCE).IsPlanar()
+            if not is_surface_planar:
                 raise ValueError("Planes can only be created from planar faces")
             properties = GProp_GProps()
             BRepGProp.SurfaceProperties_s(arg_face.wrapped, properties)
             self._origin = Vector(properties.CentreOfMass())
-            self.x_dir = (
-                Vector(arg_x_dir)
-                if arg_x_dir
-                else Vector(
-                    BRep_Tool.Surface_s(arg_face.wrapped).Position().XDirection()
-                )
-            )
+            if isinstance(surface, Geom_BoundedSurface):
+                point = gp_Pnt()
+                face_x_dir = gp_Vec()
+                tangent_v = gp_Vec()
+                surface.D1(0.5, 0.5, point, face_x_dir, tangent_v)
+            else:
+                face_x_dir = surface.Position().XDirection()
```

Comment:
The important ability to force the x direction has been removed and needs to be returned.

Reply:
I don't think that is the case, the line 2109 retains the ability to force the x_direction from the argument to the Plane constructor. All the tests pass so I suppose such an regression would have been caught by them, unless I overlooked something.